app/database.ts

```diff
@@ -274,6 +287,10 @@ export class Database_View implements ProfileSetter, ProfileGetter, EventRemover
                 return pEvent;
             }
             this.setProfile(pEvent);
+        } else if (parsedEvent.kind == NostrKind.DELETE) {
+            parsedEvent.parsedTags.e.forEach((event_id) => {
+                this.deletionEvents.set(event_id, parsedEvent);
+            });
```

Comment:
Maybe not in this PR, but have you considerred, instead of building a reverse index, we just do all the access control check here and delete the event on write here?